src/Raven.Server/Documents/Patch/JavaScriptUtils.cs

```diff
@@ -261,6 +263,17 @@ internal JsValue GetDocumentId(JsValue self, JsValue[] args)
 
         internal JsValue TranslateToJs(Engine engine, JsonOperationContext context, object o)
         {
+            if (o is TimeSeriesRetriever.TimeSeriesStreamingRetrieverResult tsrr)
```

Comment:
Or if you need a fast version:

```csharp
if (o is List<object> list)
{
    var jsArray = new Jint.Runtime.Descriptors.PropertyDescriptor[list.Count];
    for (var i = 0; i < jsArray.Length; i++)
    {
        jsArray[i] = new Jint.Runtime.Descriptors.PropertyDescriptor(TranslateToJs(engine, context, list[i]), PropertyFlag.ConfigurableEnumerableWritable);
    }

    return new ArrayInstance(engine, jsArray);
}
```

Reply:
No, in this case, we are *not* returning an array. Instead, we are taking a C# object and translating that to a format that the JS engine can work with, since the user is going to do transformations on that.